controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentIssueReporter.java

```diff
@@ -74,22 +74,29 @@ private void maintainDeploymentIssues(List<Application> applications) {
             else
                 controller().applications().setJiraIssueId(application.id(), Optional.empty());
 
-        // TODO: Do this when version.confidence is BROKEN instead?
+        // TODO: Do this when version.confidence is BROKEN instead? Or, exclude above those upgrading to BROKEN version?
         if (failingApplications.size() > 0.2 * applications.size()) {
             fileOrUpdate(manyFailingDeploymentsIssueFrom(failingApplications)); // Problems with Vespa is the most likely cause when so many deployments fail.
         }
         else {
             for (Application application : failingApplications) {
                 Issue deploymentIssue = deploymentIssueFrom(application);
+                Tenant applicationTenant = null;
                 Classification applicationOwner = null;
                 try {
-                    applicationOwner = jiraClassificationOf(ownerOf(application));
+                    applicationTenant= ownerOf(application);
+                    applicationOwner = jiraClassificationOf(applicationTenant);
                     fileFor(application, deploymentIssue.with(applicationOwner));
                 }
                 catch (RuntimeException e) { // Catch errors due to inconsistent or missing data in Sherpa, OpsDB, JIRA, and send to ourselves.
                     Pattern componentError = Pattern.compile(".*Component name '.*' is not valid.*", Pattern.DOTALL);
                     if (componentError.matcher(e.getMessage()).matches()) // Several properties seem to list invalid components, in which case we simply ignore this.
-                        fileFor(application, deploymentIssue.with(applicationOwner.withComponent(null)));
+                        fileFor(application,
+                                deploymentIssue
+                                        .with(applicationOwner.withComponent(null))
+                                        .append("\n\nNote: The 'Queue Component' field in [opsdb|https://opsdb.ops.yahoo.com/properties.php?id=" +
```

Comment:
I was too quick with this. Obviously you cannot write this error message here.

Reply:
.. nor the comment above.

Reply:
I assume you mean the error message needs to be provided by the implementation of Properties, or some such? (In addition to this being poor code style, with a template in the midst of logic ...)

Reply:
The comment comment you need to comment on, though.

Reply:
Remember: This is an open source project. It has none of these things mentioned in that comment.

I think you shouldn't bother doing something more complicated than removing the comment and removing any mention of a specific system from the message.

Reply:
I had a rework of the whole thing in mind, leaving only the detect-problems-and-report part in the DeploymentIssueReporter, and delegating the chains

PropertyId (barely ok) --OpsDB--> Classification --JIRA--> issueKey

and

IssueInfo (with assignee) --OpsDB--> UserContact escalationTarget --JIRA--> void (but escalated issue) 

to some 
```java
interface OrganizationDatabase {
    String fileFor(Application application, Issue issue);
    void escalateIssue(IssueInfo issueInfo);
}
```
or something. Sounds OK, or too much (pointless) work?

Reply:
Agree, that sounds good.